### DISCUSSION TITLE
PYTHONASYNCIODEBUG warning

### DISCUSSION THREAD

**Discussion by PaulRudin** [Kludex/uvicorn#1941] [Q&A]
I'm doing a little bit of profiling and performance checking. If I set PYTHONASYNCIODEBUG I sometimes see warnings like this:
```
 WARNING:asyncio:Executing <Task pending name='Task-36232' coro=<RequestResponseCycle.run_asgi() running at /venv/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl │
│ .py:436> wait_for=<_GatheringFuture pending cb=[Task.task_wakeup()] created at /usr/local/lib/python3.11/asyncio/tasks.py:699> cb=[set.discard()] created at /venv/lib/python3.11/s │
│ ite-packages/uvicorn/protocols/http/httptools_impl.py:313> took 0.194 seconds 
```
Presumably this means that the event loop is blocking for 0.2 of a second. Why might this be and is there anything that can be done?

**Comment by Kludex** [comment]
Well, that's the coroutine that runs your code. :eyes:

**Comment by PaulRudin** [comment]
Right, but why isn't the invoked coroutine shown in the warning? There are a couple of parametrised endpoints and I'm load testing with random requests. I suspect that the spikes coincide with garbage collection - but I might be wrong about that...

**Reply by euri10** [reply]
might be yeah, see https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task and the Important note.

I would be interesting to see if lines 313 - 314 are inverted (like in the example from the docs, callback is added after the task is added to the set, we do the opposite) it solves it
